### DISCUSSION TITLE
Можно ли развести youtube и iptv

### DISCUSSION THREAD

**Discussion by RareBirdMe** [bol-van/zapret#350] [General]
Zapret установлен на роутере (flint 2), задача была наладить, среди прочего, работу андроид тв, чтобы без танцев с бубном. Нюанс был в том, что в приложении ютуба на тв, похоже, интерфейс работает на одном протоколе, а видеопоток идёт через другой, и для мгтс нужны одновременно разные стратегии, для http и https. Zapret'ом это всё прекрасно нашлось и заработало. Пока поставил полную обработку всего трафика.

Но сломалось iptv (известный вариант через ottplayer), используемое на том же тв. Стал отслеживать и вносить в exclude-лист айпишники, набралось довольно много ip, отдельных и явно пулами. Объединил, как приведено ниже, тв заработно, но начал подтормаживать ютуб - предполагаю, что пулами зацепляет серверы ютуба. Если убрать exclude, то ютуб снова работает прекрасно.

Пока только идея вернуться от пулов к спискам, но там явно ещё будут всплывать айпишники, а тв, соответственно, прерываться. А если это действительно одно облако, то таким образом бессмысленно. Может, есть у кого мысли на этот счёт, можно ли как-то решить (естественно, автоматически, а не вручную включать/выключать списки).

<details>

<summary>Отслеженные IP iptv (очевидно, не все)</summary>

77.72.19.179
77.72.22.162
88.210.39.34
89.248.160.62

5.182.228.10
5.182.228.114
5.182.228.122
5.182.228.130
5.182.228.138
5.182.228.146
5.182.228.186
5.182.228.202
5.182.228.210
5.182.228.218
5.182.228.226
5.182.228.250

5.182.229.18
5.182.229.42
5.182.229.50

5.182.231.27
5.182.231.60
5.182.231.106
5.182.231.114
5.182.231.122
5.182.231.130
5.182.231.138
5.182.231.194
5.182.231.210
5.182.231.234

5.188.172.210

92.119.88.50
92.119.88.82
92.119.88.146
92.119.88.155
92.119.88.162
92.119.88.170
92.119.88.202
92.119.88.242

92.119.89.176
92.119.89.234

93.189.56.122
93.189.56.146
93.189.56.227

93.189.60.179

185.6.14.42
185.6.14.50
185.6.14.74

194.59.59.42
194.59.59.82
194.59.59.83
194.59.59.84
194.59.59.86
194.59.59.87
194.59.59.88
194.59.59.89
194.59.59.90
194.59.59.170

194.59.155.2
194.59.155.138
194.59.155.146
194.59.155.170
194.59.155.178
194.59.155.186
194.59.155.218
194.59.155.226
194.59.155.234
194.59.155.242
194.59.155.250

213.183.40.114
213.183.40.123
213.183.40.146
213.183.40.226

213.183.47.162

213.226.69.170
</details>

<details>

<summary>Что занесено в exclude и что, похоже, зацепляет ютуб</summary>

5.182.228.0/24
5.182.229.0/24
5.182.231.0/24
92.119.88.0/24
92.119.89.0/24
93.189.56.0/24
194.59.59.0/24
194.59.155.0/24
213.183.40.0/24
213.183.47.0/24
api.ottplayer.tv
77.72.19.179
77.72.22.162
88.210.39.34
89.248.160.62
</details>

**Comment by bol-van** [comment]
чем плох хостлист ? он не связан с ип

**Comment by RareBirdMe** [comment]
Если пойти через hostlist exclude iptv:
- можно внести ottplayer.tv, ottclub.xyz, с них скачиваются плей-листы m3u8;
- в плей-листах фрагменты лежат сразу по ip (для примера приложил файл) - получается, не вариант.

Если через include, то, конечно, возникает вопрос списков. Со стандартно скачиваемым include-списком из 150k строк flint 2 c nftables, похоже, не справляется - как и предупреждали. Вести свой, маленький, руками? Но в целом вариант рабочий, спасибо.

[index.zip](https://github.com/user-attachments/files/16841330/index.zip)

**Reply by bol-van** [reply]
смерть роутерам лишь от больших nfset
для хостлистов 150к  нужно 128  ram

в хостлист можно вносить и ип
в заголовке   host:
будет ip
хотя лучше проверить шарком
может быть  ip:80

еще если в стратегии есть ttl нужно убедиться, что iptv не ближе по хопам
если так нужен доп ограничитель fooling

**Reply by RareBirdMe** [reply]
Отчитываюсь по вашим рекомендациям:
1. Удалось запустить с "большим" хост-листом (для меня, видимо, проще каждый раз перезапускать через install easy). Примерно 60 Мб памяти отъело на 3 процесса (ipv4). На 6 процессов будет как раз 128.
2. Попробовал альтернативный путь - убрал хостлисты, добавил в стратегии на http и https c ttl --dpi-desync-fooling=md5sig. Ещё посмотрю, но на первый взгляд всё ок. Iptv не ломается. То что нужно, работает. 
Таким образом, даже два рабочих варианта получилось, просто отлично.

Кстати, возможно, стоит в blockcheck добавить, чтобы в найденных стратегиях с ttl сразу добавлялся md5sig (если оно безвредно).

**Reply by bol-van** [reply]
блокчек не для простоты, а для исследования  dpi
это менять не планируется
